### PR TITLE
Chinese remainder theorem dart

### LINKED ISSUE #2481: Chinese Remainder Theorem
- [x] C - @Anushka-Rajvanshi
- [x] C++
- [x] C#
- [x] CoffeeScript
- [x] Dart - @Rukmini-Meda
- [x] Go - @jedicakelord
- [x] Java
- [x] JS
- [ ] Kotlin
- [x] Php - @sukritishah15
- [x] Python
- [x] Readme - @chandrikadeb7
- [ ] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by Rukmini-Meda**
Fixes #2481 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Chinese Remainder Theorem is implemented in Dart

#### Languages Used:
- Dart

#### Files Added:
- Chinese_Remainder_Theorem.dart

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by Rukmini-Meda** [comment]
I have implemented Chinese Remainder Theorem in Dart following all the required coding conventions. Please review and merge. Thank you.

File: Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.dart [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        result += temp * (remainders[i]) * modular_inverse(temp,numbers[i]);
+    }
+    
+    result = result % product;
+
+    // Printing result
+    print("The value of 'x' by Chinese Remainder Theorem is $result.");
+
+    /*
+        Sample Input
```
Comment (singh-shreya6) [inline Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.dart:105]: Please combine this input output into one multi comment block.

File: Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.dart [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+    /*
+        Sample Output
+        
+        The value of 'x' by Chinese Remainder Theorem is 298.
+
+    */
+
+}
+
```
Comment (singh-shreya6) [inline Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.dart:122]: Please keep only one blank line at the end of file.

File: Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.dart [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        1 2 3 4 5
+
+    */
+
+    /*
+        Sample Output
+        
+        The value of 'x' by Chinese Remainder Theorem is 298.
+
+    */
```
Comment (singh-shreya6) [inline Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.dart:119]: Also it is best to remove sample input and output from main. Close the main method and add these at end of file as a single multi line comment.

**Comment by Rukmini-Meda** [comment]
I have made the changes as you said. Please review and merge. I have a doubt. Can I know the significance of the new line at the end of the file? Thank you.

**Pull request merged**